cmake/onnxruntime_providers_migraphx.cmake

```diff
@@ -33,26 +23,24 @@
   find_package(hip REQUIRED)
   find_package(migraphx REQUIRED PATHS ${AMD_MIGRAPHX_HOME})
 
-  set(migraphx_libs migraphx::c hip::host)
-
   file(GLOB_RECURSE onnxruntime_providers_migraphx_cc_srcs CONFIGURE_DEPENDS
     "${ONNXRUNTIME_ROOT}/core/providers/migraphx/*.h"
     "${ONNXRUNTIME_ROOT}/core/providers/migraphx/*.cc"
     "${ONNXRUNTIME_ROOT}/core/providers/shared_library/*.h"
     "${ONNXRUNTIME_ROOT}/core/providers/shared_library/*.cc"
   )
   source_group(TREE ${ONNXRUNTIME_ROOT}/core FILES ${onnxruntime_providers_migraphx_cc_srcs})
-  onnxruntime_add_shared_library_module(onnxruntime_providers_migraphx ${onnxruntime_providers_migraphx_cc_srcs})
+  onnxruntime_add_shared_library(onnxruntime_providers_migraphx ${onnxruntime_providers_migraphx_cc_srcs})
```

Comment:
Why did you change it from onnxruntime_add_shared_library_module to onnxruntime_add_shared_library ? A Module Library is a plugin that may not be linked by other targets, but may be dynamically loaded at runtime using dlopen-like functionality.

Reply:
We have a use case which does link to this. Hence why this was added and integrated back on the ROCm side. Let me know whats the optimal way to handle this. Looks like I'll need to handle the integration work on this. Appreciate your comments sand insights btw @snnn

Reply:
I don't disagree with you. Unfortunately this was pushed from windows side changes since the original interface was changed. Seems like a better solution is to keep the old flags then (save/load) and add the newer one (model cache) so that we don't break abi compatibility then and just extend it?

Reply:
We changed to `onnxruntime_add_shared_library` because `onnxruntime_add_shared_library_module` puts the provider DLL in `lib` directory when installing with `cmake --install ...`, and we need the DLL in `bin` directory, and `onnxruntime_add_shared_library` does precisely that. The side effects are the header and interface library installed alongside.